### PR TITLE
[GraphBolt][CUDA] Warning there are multiple `CopyTo`.

### PR DESCRIPTION
**Pull request by mfbalin**
## Description
@frozenbugs When there are accidentally multiple `CopyTo` ops, each will have a prefetcher etc. making overall performance worse. Do you think we should add a check inside `gb.DataLoader` to see if there are multiple `CopyTo` in the datapipe graph?

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/dataloader.py [hunk 1]
```diff
@@ -226,6 +226,9 @@ def __init__(
         # separate thread.
         if torch.cuda.is_available():
             copiers = dp_utils.find_dps(datapipe_graph, CopyTo)
+            assert (
```
Comment (frozenbugs) [inline python/dgl/graphbolt/dataloader.py:229]: Do we believe this is always true? Is it possible that when user customize graphbolt's feature fetcher, such that they have copyto to copy sampled graph first and then copy feature feature later?

Comment (mfbalin) [reply python/dgl/graphbolt/dataloader.py:229]: That would happen only if they have 2 branches that they are combining, not a single chain of datapipes İ feel. Even then, they can combine the chains and then use a single copyto.

Comment (mfbalin) [reply python/dgl/graphbolt/dataloader.py:229]: Our featurefetcher now copies features to the gpu when ids are on the gpu for all our supported feature classes.

Comment (frozenbugs) [reply python/dgl/graphbolt/dataloader.py:229]: Probably just a warning is enough, don't make it a deal breaker.

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**